### PR TITLE
Refactored optional position parameter passing

### PR DESCRIPTION
**Pull request by biodranik**
Related https://github.com/organicmaps/organicmaps/pull/9002

It makes the meaning of passed parameters more obvious.

### CODE REVIEW

File: drape_frontend/message_subclasses.hpp [hunk 1]
```diff
@@ -405,28 +406,26 @@ class SetAddNewPlaceModeMessage : public Message
...
 public:
   SetAddNewPlaceModeMessage(bool enable, std::vector<m2::TriangleD> && boundArea,
-                            bool enableKineticScroll, bool hasPosition, m2::PointD const & position)
+                            bool enableKineticScroll, m2::PointD const * optionalPosition)
     : m_enable(enable)
     , m_boundArea(std::move(boundArea))
     , m_enableKineticScroll(enableKineticScroll)
-    , m_hasPosition(hasPosition)
-    , m_position(position)
+  , m_position(optionalPosition ? *optionalPosition : decltype(m_position){})
```
Comment (vng) [inline drape_frontend/message_subclasses.hpp:413]: nit: I'd prefer in ctor:
```
if (optionalPosition)
  m_position = *optionalPosition;
```

File: drape_frontend/message_subclasses.hpp [hunk 2]
```diff
@@ -405,28 +406,26 @@ class SetAddNewPlaceModeMessage : public Message
...
 
   Type GetType() const override { return Type::SetAddNewPlaceMode; }
 
   std::vector<m2::TriangleD> && AcceptBoundArea() { return std::move(m_boundArea); }
   bool IsEnabled() const { return m_enable; }
   bool IsKineticScrollEnabled() const { return m_enableKineticScroll; }
-  bool HasPosition() const { return m_hasPosition; }
-  m2::PointD const & GetPosition() const { return m_position; }
+  bool HasPosition() const { return m_position.has_value(); }
+  m2::PointD GetPosition() const { return *m_position; }
```
Comment (vng) [inline drape_frontend/message_subclasses.hpp:422]: Add ASSERT here?

**Review by vng — APPROVED** [review]

File: drape_frontend/message_subclasses.hpp (hunk 1 above)
Comment (biodranik) [reply drape_frontend/message_subclasses.hpp:413]: Ok. The initial idea was to make some variables const, the if doesn't allow that.

File: drape_frontend/message_subclasses.hpp [hunk 3]
```diff
@@ -405,28 +406,27 @@ class SetAddNewPlaceModeMessage : public Message
...
+  }
 
   Type GetType() const override { return Type::SetAddNewPlaceMode; }
 
   std::vector<m2::TriangleD> && AcceptBoundArea() { return std::move(m_boundArea); }
   bool IsEnabled() const { return m_enable; }
   bool IsKineticScrollEnabled() const { return m_enableKineticScroll; }
-  bool HasPosition() const { return m_hasPosition; }
-  m2::PointD const & GetPosition() const { return m_position; }
+  auto const & GetOptionalPosition() const { return m_position; }
```
Comment (biodranik) [inline drape_frontend/message_subclasses.hpp:423]: @vng is it better now?

**Pull request merged**